src/languages/cs.js

```diff
@@ -134,15 +137,23 @@ function(hljs) {
         ]
       },
       {
+        // [Attributes("")]
+        className: 'title',
+        begin: '^\\s*\\[', excludeBegin: true, end: '[\\(\\]]', excludeEnd: true,
+        contains: [
+          STRING
+        ]
+      },
+      {
         // Expression keywords prevent 'keyword Name(...)' from being
         // recognized as a function definition
         beginKeywords: 'new return throw await',
         relevance: 0
       },
       {
         className: 'function',
-        begin: '(' + TYPE_IDENT_RE + '\\s+)+' + hljs.IDENT_RE + '\\s*\\(', returnBegin: true, end: /[{;=]/,
-        excludeEnd: true,
+        begin: '(' + TYPE_IDENT_RE + '\\s+)+[A-Z]\\w*\\s*\\(', returnBegin: true,
+        end: /[{;=]/, excludeEnd: true,
```

Comment:
This change fix the `else if()` taken as a function.
In your use case, the function name starts with a lower-case letter.
In C#, naming guideline advises to use pascal case for method names.
So it seemed to me a good way to fix the `else if` issue.